File: iut/TP3_TP4/chute.c
#include <math.h>
#include <stdio.h>

void chute(double hauteur)
{
    double g = 9.80665;
    double res;
    int i,test;

    i = 0;
    res = 2 * hauteur;
    res = res / g;
    res = sqrt(res);
    test = res;
    printf("%i %f\n", test, res);
    while (test != 0)
    {
        printf("%is : %f m\n",i, hauteur);
        test = test - 1;
        i++;
        hauteur = hauteur - (0.5*g*i*i);
    }
    printf("%is : boum",i);
}

int main()
{
    chute(100);
    return 0;
}

File: iut/TP3_TP4/chute.py
import math

def chute(hauteur):
    g = 9.80665
    i = 0
    res = 2 * hauteur
    res = res / g
    res = math.sqrt(res)
    test = res
    
    while (test >= 0):
        print(f"{i}s : {hauteur} m")
        test = test - 1
        i+=1
        hauteur = hauteur - (0.5*g*i*i)
    print(f"{i}s : boum")

chute(100)